src/background/services/paymentSession.ts

```diff
@@ -84,56 +84,46 @@ export class PaymentSession {
     Object.assign(this, this.deps);
   }
 
-  #adjustAmountLastRate: AmountValue;
-  #adjustAmountController = new AbortController();
-  #adjustAmountPromise: null | Promise<void> = null;
-  adjustAmount(rate: AmountValue): Promise<void> {
-    if (this.#adjustAmountLastRate && rate !== this.#adjustAmountLastRate) {
-      this.#adjustAmountController.abort(
-        new DOMException(
-          `Aborting existing probing for rate=${this.#adjustAmountLastRate}`,
-          'AbortError',
-        ),
-      );
-      this.#adjustAmountController = new AbortController();
-      this.#adjustAmountPromise = null;
-    }
-
-    this.#adjustAmountLastRate = rate;
-    this.#adjustAmountPromise ??= this._adjustAmount(
-      rate,
-      this.#adjustAmountController.signal,
-    );
-
-    return this.#adjustAmountPromise;
+  // We keep setting #minSendAmount to non-zero values as we probe. Instead of
+  // checking #minSendAmount > 0, use this boolean to know if probing completed.
+  #minSendAmountFound = false;
+  #minSendAmount = 0n;
+  #minSendAmountPromise: ReturnType<typeof this._findMinSendAmount>;
+
+  findMinSendAmount(): Promise<void> {
+    this.#minSendAmountPromise ??= this._findMinSendAmount();
```

Comment:
This ensures we call `_findMinSendAmount()` only once per `PaymentSession` instance.